akka-remote/src/test/scala/akka/remote/artery/SendQueueSpec.scala

```diff
@@ -147,6 +148,67 @@ class SendQueueSpec extends AkkaSpec("akka.actor.serialize-messages = off") with
       downstream.cancel()
     }
 
+    "deliver first message" in {
+
+      def test(f: (ManyToOneConcurrentArrayQueue[String], SendQueue.QueueValue[String], TestSubscriber.Probe[String]) ⇒ Unit): Unit = {
+
+        (1 to 100).foreach { n ⇒
+          val queue = new ManyToOneConcurrentArrayQueue[String](16)
```

Comment:
But this queue isn't really used anywhere except for passing it to `f`?

Reply:
The test is exercising the different combinations that the queue and the async callback can be used. One way is to add to the queue before injecting the queue into the stage.

Reply:
Aah, now I saw the call where it is injected, must have missed that last time. Thanks for clarifying.